src/components/ProductListing/Products/Products.jsx

```diff
@@ -1,13 +1,33 @@
+import { useState, useEffect } from "react";
 import { Product } from "../Product/Product";
+import axios from "axios";
 
 export const Products = () => {
+  const [products, setProducts] = useState([]);
+
+  const listProducts = async () => {
+    try {
+      const response = await axios.get("/api/products");
+      setProducts(response.data.products);
```

Comment:
Here first check the response status and if it is 200 then only setProducts otherwise throw an error.

Reply:
Done. Thank you for this suggestion Atul!